File: vol1/ch3/LeetCode739.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n=len(temperatures)
        ans=[0]*n
        st=[]
        for i in range(0,n):
            while st and temperatures[st[-1]]<temperatures[i]:
                prei=st[-1]
                ans[prei]=i-prei
                st.pop()
            st.append(i)
        return ans 

File: vol1/ch3/LeetCode739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n=temperatures.size();
        vector<int> ans(n,0);
        stack<int> st;
        for (int i=0;i<n;i++) {
            while(!st.empty() && temperatures[st.top()]<temperatures[i]) {
                int prei=st.top();
                ans[prei]=i-prei;
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};

